format-common/src/main/java/io/cdap/plugin/format/input/PathTrackingInputFormatProvider.java

```diff
@@ -63,6 +65,31 @@ protected void validate() {
     // no-op
   }
 
+  protected Schema addPathField(Schema dataSchema, FailureCollector failureCollector) {
+    return addPathField(failureCollector, dataSchema, conf.getPathField());
+  }
+
+  public static Schema addPathField(FailureCollector failureCollector, Schema dataSchema, String pathField) {
+    if (pathField == null) {
+      return dataSchema;
+    }
+    if (dataSchema == null) {
```

Comment:
our convention is to assume method arguments cannot be null unless they are annotated with `@Nullable`. It doesn't look like anything is passing in null for the schema, so this check can be removed.

Reply:
Pathfield can be null if not passed

Reply:
Removed null check for dataschema